src/infer_request.cc

```diff
@@ -420,10 +420,23 @@ InferenceRequest::Run(std::unique_ptr<InferenceRequest>& request)
   return status;
 }
 
+FailureReason stringToFailureReason(const std::string& error_type) {
+    if (error_type == "REJECTED") {
```

Comment:
@Tabrizian what do you think about "Rejected" vs "Timeout" for this category of failures?
```
nv_inference_request_failure{model="llama-3-8b-instruct",version="1",reason="TIMEOUT"} 7
```
vs.
```
nv_inference_request_failure{model="llama-3-8b-instruct",version="1",reason="REJECTED"} 7
```

Originally I opted for "REJECTED" because of:
```
  enum TimeoutAction {
    REJECT = 0;
    DELAY = 1;
}
```
and only the REJECTED timeout would be a "failure", but TIMEOUT may be more intuitive.

Reply:
Is it possible to merge the [error types](https://github.com/triton-inference-server/common/blob/00b3a71519e32e3bc954e9f0d067e155ef8f1a6c/include/triton/common/error.h#L39-L49) with failure reason data structure? I think ideally we want the backend to be able to report these categories as well. I may not have to full context on this feature so feel free to correct my understanding. I slightly prefer timeout since it could be used for connection timeout as well.